File: Python/Program/Course.py
class Course:
    # Kelas Course di sini memiliki hubungan Komposisi dengan Kelas Program Studi.
    # Hubungan Komposisi dengan Kelas Course digunakan karena Course atau Mata Kuliah merupakan part-of Program Studi.
    # Hal tersebut bermakna ketika Program Studi dihapus, maka Course juga akan terhapus.

    # ATRIBUT
    __nama_matakuliah = ""
    __jumlahSKS = 0

    # METHODS
    # Constructor
    def __init__(self, nama, jumlah):
        self.__nama_matakuliah = nama
        self.__jumlahSKS = jumlah

    # Setter and Getter
    def getNamaMatkul(self):
        return self.__nama_matakuliah

    def setNamaMatkul(self, matkul):
        self.__nama_matakuliah = matkul

    def getJumlahSKS(self):
        return self.__jumlahSKS
    
    def setJumlahSKS(self, jmlSKS):
        self.__jumlahSKS = jmlSKS

File: CPP/Program/course.cpp
#include "header.hh"

// == Kelas Course ==

Course::Course()
{
    this->nama_matakuliah = "";
    this->jumlahSKS = 2;
}
Course::Course(string matkul, int jmlSKS)
{
    this->nama_matakuliah = matkul;
    this->jumlahSKS = jmlSKS;
}

// Getter and Setter
string Course::getNamaMatkul()
{
    return this->nama_matakuliah;
}
int Course::getJumlahSKS()
{
    return this->jumlahSKS;
}
void Course::setNamaMatkul(string matkul)
{
    this->nama_matakuliah = matkul;
}
void Course::setJumlahSKS(int jmlSKS)
{
    this->jumlahSKS = jmlSKS;
}

// Destructor
Course::~Course()
{
}
